File: lever/src/main/java/com/pin/lever/FetchAction.java
package com.pin.lever;

import android.graphics.Bitmap;

class FetchAction extends Action<Object> {

    private final Object target;
    private Callback callback;

    FetchAction(Lever lever, Request data, int memoryPolicy, int networkPolicy, Object tag,
                String key, Callback callback) {
        super(lever, null, data, memoryPolicy, networkPolicy, 0, null, key, tag, false);
        this.target = new Object();
        this.callback = callback;
    }

    @Override void complete(Bitmap result, Lever.LoadedFrom from) {
        if (callback != null) {
            callback.onSuccess();
        }
    }

    @Override void error(Exception e) {
        if (callback != null) {
            callback.onError(e);
        }
    }

    @Override void cancel() {
        super.cancel();
        callback = null;
    }

    @Override Object getTarget() {
        return target;
    }
}

File: lever/src/main/java/com/pin/lever/FetchAction.kt
package com.pin.lever

import android.graphics.Bitmap
import com.pin.lever.Lever.LoadedFrom

internal class FetchAction(
    lever: Lever?,
    data: Request?,
    memoryPolicy: Int,
    networkPolicy: Int,
    tag: Any?,
    key: String?,
    callback: Callback?
) : Action<Any?>(
    lever,
    null,
    data,
    memoryPolicy,
    networkPolicy,
    0,
    null,
    key,
    tag,
    false
) {
    private val targetAny: Any
    private var callback: Callback?
    public override fun complete(result: Bitmap?, from: LoadedFrom?) {
        if (callback != null) {
            callback!!.onSuccess()
        }
    }

    public override fun error(e: Exception?) {
        if (callback != null) {
            callback!!.onError(e!!)
        }
    }

    public override fun cancel() {
        super.cancel()
        callback = null
    }

    public override fun getTarget(): Any {
        return targetAny
    }

    init {
        this.targetAny = Any()
        this.callback = callback
    }
}